src/Core/Geographic/Extent.js

```diff
@@ -263,4 +271,33 @@ Extent.prototype.intersect = function intersect(bbox) {
              this.north() <= other.south(this._internalStorageUnit));
 };
 
+Extent.prototype.set = function set(west, east, south, north) {
+    this._values[CARDINAL.WEST] = west;
+    this._values[CARDINAL.EAST] = east;
+    this._values[CARDINAL.SOUTH] = south;
+    this._values[CARDINAL.NORTH] = north;
+};
+
+Extent.prototype.add = function add(extent) {
```

Comment:
You should check that `extent.crs()` is the same as this.crs() and throw otherwise.